stdlib/join/join_fn.go

```diff
@@ -0,0 +1,304 @@
+package join
+
+import (
+	"context"
+	"sort"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/compiler"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/interpreter"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+type JoinFn struct {
+	fn       *semantic.FunctionExpression
+	scope    compiler.Scope
+	compiled compiler.Func
+	args     values.Object
+	schema   []flux.ColMeta
+	ltyp     *semantic.MonoType
+	rtyp     *semantic.MonoType
+}
```

Comment:
Is `JoinFn` for the `on` or the `as` parameter? It would be good to have a comment here that explains.